<p></p>
<h2>Noncompliant Code Example (AsyncAPI 2)</h2>
<pre>
asyncapi: '2.4.0'
info:
  title: Non-Compliant API
  version: '1.0.0'
  description: This API does not follow the recommendation to add servers and channels to components.
  license:
    name: Apache 2.0
    url: https://www.apache.org/licenses/LICENSE-2.0.html
servers: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{AAR041: It is recommend to add the servers and channels to component}}</span>
  production:
    url: api.example.com
    protocol: https
    description: Production server
channels: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{AAR041: It is recommend to add the servers and channels to component}}</span>
  pets:
    description: Channel for managing pet data
    subscribe:
      summary: Subscribe to pet updates
      operationId: receivePetUpdates
      message:
        contentType: application/json
        payload:
          type: object
          properties:
            id:
              type: string
              description: ID of the pet
            name:
              type: string
              description: Name of the pet

</pre>
<h2>Compliant Solution (AsyncAPI 2)</h2>
<pre>
asyncapi: '2.4.0'
info:
  title: Compliant API
  version: '1.0.0'
  description: This API follows the recommendation to add servers and channels to components.
  license:
    name: Apache 2.0
    url: https://www.apache.org/licenses/LICENSE-2.0.html

components:
  servers:
    production:
      url: api.example.com
      protocol: https
      description: Production server
  channels:
    pets:
      description: Channel for managing pet data
      subscribe:
        summary: Subscribe to pet updates
        operationId: receivePetUpdates
        message:
          contentType: application/json
          payload:
            type: object
            properties:
              id:
                type: string
                description: ID of the pet
              name:
                type: string
                description: Name of the pet

</pre>