<p></p>
<h2>Noncompliant Code Example (AsyncAPI 2)</h2>
<pre>
asyncapi: "2.0.0"
info:
  title: Bad API
  version: "1.0.0"
components:
  messages:
    someMessage:
      payload:
        type: string
      headers:
        type: object
      examples: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{All examples in message object should follow payload and headers schemas}}</span>
        - payload: 2137
          headers: someHeader
</pre>
<h2>Compliant Solution (AsyncAPI 2)</h2>
<pre>
asyncapi: "2.0.0"
info:
  title: Good API
  version: "1.0.0"
components:
  messages:
    someMessage:
      payload:
        type: string
      headers:
        type: object
      examples:
        - payload: foobar
          headers:
            someHeader: someValue
</pre>