<p></p>
<h2>Noncompliant Code Example (AsyncAPI 2)</h2>
<pre>
asyncapi: '2.4.0'
info:
  title: Example API
  version: '1.0.0'
servers:
  production:
    url: mqtt://broker.example.com
    protocol: mqtt
channels:
  /example-channel:
    description: Example channel for testing.
    bindings: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{AAR037: You must specify the version of the binding}}</span>
      mqtt:
        qos: 1
    subscribe:
      summary: Example subscription
      operationId: exampleSubscription
      message:
        contentType: application/json
</pre>
<h2>Compliant Solution (AsyncAPI 2)</h2>
<pre>
    asyncapi: '2.4.0'
    info:
      title: Example API
      version: '1.0.0'
    servers:
      production:
        url: mqtt://broker.example.com
        protocol: mqtt
    channels:
      /example-channel:
        description: Example channel for testing.
        bindings:
          mqtt:
            qos: 1
            version: 1.0.0
        subscribe:
          summary: Example subscription
          operationId: exampleSubscription
          message:
            contentType: application/json
    
</pre>