<p>Numeric parameters should have minimum, maximum, or format restriction</p>
<h2>Noncompliant Code Example (AsyncAPI 2)</h2>
<pre>
    asyncapi: '2.4.0'
    info:
      title: Numeric Parameter Restrictions Example API
      version: '1.0.0'
      description: API de ejemplo que incluye restricciones para parámetros numéricos.
    
    channels:
      /sensor-data:
        description: Canal para publicar datos de sensores.
        publish:
          operationId: sendSensorData
          summary: Publica datos recogidos por un sensor.
          message:
            contentType: application/json
            payload:
              type: object
              properties: 
                temperature:
                  type: number <span class="error-info" style="color: #FD8E18;">#Noncompliant {{AAR032: Numeric parameters should have minimum, maximum, or format restriction}}</span>
                  description: Temperatura medida por el sensor.
                sensorId: <span class="error-info" style="color: #FD8E18;">#Noncompliant {{AAR032: Numeric parameters should have minimum, maximum, or format restriction}}</span>
                  type: integer
                  description: ID único del sensor.    
</pre>
<h2>Compliant Solution (AsyncAPI 2)</h2>
<pre>
asyncapi: '2.4.0'
info:
  title: Numeric Parameter Restrictions Example API
  version: '1.0.0'
  description: API de ejemplo que incluye restricciones para parámetros numéricos.

channels:
  /sensor-data:
    description: Canal para publicar datos de sensores.
    publish:
      operationId: sendSensorData
      summary: Publica datos recogidos por un sensor.
      message:
        contentType: application/json
        payload:
          type: object
          properties: 
            humidity:
              type: number
              description: Humedad medida por el sensor.
              minimum: 0
              maximum: 100
            pressure:
              type: number
              description: Presión atmosférica medida.
              minimum: 900
              maximum: 1100
</pre>